beacon_node/network/src/nat.rs

```diff
@@ -3,225 +3,55 @@
-                        debug!(log, "UPnP port in use, attempting to remap"; "protocol" => protocol_string, "port" => socket.port());
-                        match gateway.remove_port(protocol, socket.port()) {
-                            Ok(()) => {
-                                debug!(log, "UPnP Removed port mapping"; "protocol" => protocol_string,  "port" => socket.port())
-                            }
-                            Err(e) => {
-                                debug!(log, "UPnP Port remove failure"; "protocol" => protocol_string, "port" => socket.port(), "error" => %e);
-                                return Err(());
-                            }
-                        }
-                    }
-                    e => {
-                        info!(log, "UPnP TCP route not set"; "error" => %e);
-                        return Err(());
-                    }
-                }
-            }
-            Ok(_) => {
-                return Ok(());
-            }
-        }
+    if is_private {
+        bail!("Gateway's external address is a private address: {external_address}");
     }
-    Err(())
-}
 
-/// Removes the specified TCP and UDP port mappings.
-pub fn remove_mappings(mappings: &EstablishedUPnPMappings, log: &slog::Logger) {
-    if mappings.is_some() {
-        debug!(log, "Removing UPnP port mappings");
-        match igd::search_gateway(Default::default()) {
-            Ok(gateway) => {
-                if let Some(tcp_port) = mappings.tcp_port {
-                    match gateway.remove_port(igd::PortMappingProtocol::TCP, tcp_port) {
-                        Ok(()) => debug!(log, "UPnP Removed TCP port mapping"; "port" => tcp_port),
-                        Err(e) => {
-                            debug!(log, "UPnP Failed to remove TCP port mapping"; "port" => tcp_port, "error" => %e)
-                        }
-                    }
-                }
-                for udp_port in mappings.udp_ports() {
-                    match gateway.remove_port(igd::PortMappingProtocol::UDP, *udp_port) {
-                        Ok(()) => debug!(log, "UPnP Removed UDP port mapping"; "port" => udp_port),
-                        Err(e) => {
-                            debug!(log, "UPnP Failed to remove UDP port mapping"; "port" => udp_port, "error" => %e)
-                        }
-                    }
-                }
-            }
-            Err(e) => debug!(log, "UPnP failed to remove mappings"; "error" => %e),
-        }
+    loop {
+        gateway
+            .add_port(
+                PortMappingProtocol::UDP,
+                port,
+                SocketAddr::new(IpAddr::V4(addr), port),
+                MAPPING_DURATION,
+                "Lighthouse Discovery port",
+            )
+            .await
+            .context("Could not UPnP map port: {port} on the gateway")?;
+        info!(log, "Discovery UPnP port mapped"; "port" => %port);
```

Comment:
We have a few info logs at the moment. 
We have one from creating in libp2p, one from creating discovery and then the results from both. 

I think its probably safe to assume if libp2p works, then discovery probably will to. Maybe we make the discovery logs `debug`?

Reply:
yup, makes sense Age. Updated!